src/__tests__/extended-id.test.ts

```diff
@@ -37,3 +37,88 @@ describe('pattern matcher', () => {
     })
   })
 })
+
+describe('interpretIdentifier', () => {
+  const pubKey = '0x02b97c30de767f084ce3080168ee293053ba33b235d7116a3263d29f1450936b71'
+  const checksumAddress = '0xC662e6c5F91B9FcD22D7FcafC80Cf8b640aed247'
+
+  it('parses ethereumAddress', () => {
+    const { address, publicKey, network } = interpretIdentifier(checksumAddress.toLowerCase())
+    expect(address).toEqual(checksumAddress)
+    expect(publicKey).toBeUndefined()
+    expect(network).toBeUndefined()
```

Comment:
why is this undefined? shouldn't it be mainnet? applies to the other tests as well.

Reply:
yeah, perhaps I should have been more clear about the intent of this method.
It's a helper method that needs to split a string into components, and in case one of the components is a public key, to also compute the address.

The logic to decide whether the network is mainnet when not specified is in a different part of the code.

So, in the general case, yes, it should be mainnet, but not for this method. Here it's expected to not be defined.